src/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../types';
+
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
```

Comment:
Here you're defining the type of the post you receive as props to be an object with type Post, and it is being used in the components as so, but actually in the App.tsx up there, you're passing the post as a number, this is stoping you of receiving the post information from the post the user clicked to open.